src/dataServices/dynamoDbDataService.ts

```diff
@@ -94,13 +98,26 @@ export class DynamoDbDataService implements Persistence, BulkDataAccess {
 
     async createResource(request: CreateResourceRequest) {
         const { resourceType, resource, id } = request;
+
         const vid = 1;
         let item = resource;
         item.resourceType = resourceType;
         item.meta = generateMeta(vid.toString());
 
         const params = DynamoDbParamBuilder.buildPutAvailableItemParam(item, id || uuidv4(), vid);
-        await this.dynamoDb.putItem(params).promise();
+        try {
+            await this.dynamoDb.putItem(params).promise();
+        } catch (e) {
+            if (e.code === 'ConditionalCheckFailedException') {
+                // It is highly unlikely that the autogenerated id will collide with a preexisting id, but adding that case
+                // to cover all bases
+                const errorMessage = id
+                    ? `Unable to create resource with id ${id}, because resource already exists`
+                    : 'Auto generated id matched an existing id, please try again';

```

Comment:
This doesn't look right.

This error is being mapped to a 400, which is not correct. There's nothing wrong with the client request.
This sounds like a 500 error, it is the servers job to do what it takes to generate a unique id when creating resources. 

This is not a message that we want to show to customers. In general catching this exception is not worth it since the probability of uuid collisions is  **extremely** low. you COULD retry with a new uuid, but It's often ok to just let it be a 500 error that happens once in a million years.

Reply:
Changing the code to throw an Error, that'll be eventually mapped to a 500 with an 'Internal Server Error'.

Now that we don't allow a user to provide an `id` anymore, if there is a `ConditionalCheckFailedException`, it'll be because the randomly generated id match a preexisting id. Like @carvantes mentioned, the chance of this happening is extremely low.